src/state/actions/index.js

```diff
@@ -3,8 +3,33 @@
 // Actions should be focused to a single purpose.
 // You can have multiple action creators per file if it makes sense to the purpose those action creators are serving.
 // Declare action TYPES at the top of the file
+import { useMemo } from 'react';
+import { getRole } from '../../api/index';
 
 // USER ACTIONS
+export const authenticateUser = authService => {
+  return dispatch => {
+    dispatch(fetchStart());
+    const [memoAuthService] = useMemo(() => [authService], []);
+    memoAuthService
+      .getUser()
+      .then(async info => {
+        const role_id = await getRole(info.sub);
+        dispatch(setUserInfo({ ...info, role: role_id }));
+      })
+      .then(() => {
+        dispatch(getUserProfile());
+      })
+      .catch(err => {
+        console.log(err);
```

Comment:
Could we handle this error in state? That way, if there's an error connecting to the backend, we could let the user know in a friendly way.